lib/reek/spec/should_reek_only_of.rb

```diff
@@ -15,25 +15,53 @@ def matches?(actual)
       def matches_examiner?(examiner)
         @examiner = examiner
         @warnings = @examiner.smells
-        @warnings.length == 1 && @warnings[0].matches?(@klass, @patterns)
+        return false unless @warnings.length == 1
+        @warning = @warnings[0]
+        no_smell_details_given? ? match_without_smell_details? : match_with_smell_details?
       end
 
       def failure_message
         rpt = Cli::Report::Formatter.format_list(@warnings)
-        "Expected #{@examiner.description} to reek only of #{@klass}, but got:\n#{rpt}"
+        "Expected #{@examiner.description} to reek only of #{@smell_category}, but got:\n#{rpt}"
       end
 
       def failure_message_when_negated
-        "Expected #{@examiner.description} not to reek only of #{@klass}, but it did"
+        "Expected #{@examiner.description} not to reek only of #{@smell_category}, but it did"
+      end
+
+      private
+
+      def match_without_smell_details?
+        @warning.matches?(@smell_category)
+      end
+
+      def match_with_smell_details?
+        @smell_details_list.map do |smell_detail|
+          @warning.matches?(@smell_category, smell_detail)
+        end.all?
       end
     end
 
     #
-    # As for reek_of, but the matched smell warning must be the only warning of
-    # any kind in the target source code's Reek report.
+    # Checks the target source code for instances of +smell_category+,
+    # and returns +true+ only if it can find one and only one of them that matches.
+    # Additionally you can be more specific and pass in a list of "smell_details" you
+    # want to check for as well e.g. "name" or "count" (see the examples below).
+    #
+    # smell_category - The smell category we check for
+    # smells_details_list - An AoH with each hash containing smell_warning parameters
+    #
+    # Examples
+    #
+    #   reek_only_of(UncommunicativeParameterName, name: 'x2')
+    #   reek_only_of(DataClump, count: 3)
+    #
+    # Examples from a real spec
+    #
+    #   expect(src).to reek_only_of(:TooManyStatements,  count: 6)
     #
-    def reek_only_of(smell_category, *patterns)
-      ShouldReekOnlyOf.new(smell_category, patterns)
+    def reek_only_of(smell_category, *smell_details_list)
```

Comment:
Actually your comment still confuses me. Shouldn't this comment be rather one line below and say:

>> since we expect exactly one array argument.

?

I think we're misunderstanding each other here. 
It **was** intended that you can do:

```Ruby
expect(src).to reek_of(:UncommunicativeVariableName, { name: x1 }, { name: x2 }, { name: x3 }, ....
```

This is how we use it in the specs and that's why the tomdoc from above says:

>> smells_details_list - An AoH with each hash containing smell_warning parameters

I do agree though that the second splat in the line below is unnecessary and thus confusing.

I improved that in https://github.com/troessner/reek/commit/56057e1202ed8a1792dcb8bd856493a6ebfe95b9
Was this what you had in mind?